common/src/main/java/org/opensearch/ml/common/connector/MLPostProcessFunction.java

```diff
@@ -18,38 +18,62 @@ public class MLPostProcessFunction {
 
     public static final String COHERE_EMBEDDING = "connector.post_process.cohere.embedding";
     public static final String OPENAI_EMBEDDING = "connector.post_process.openai.embedding";
-
+    public static final String BEDROCK_EMBEDDING = "connector.post_process.bedrock.embedding";
     public static final String DEFAULT_EMBEDDING = "connector.post_process.default.embedding";
 
     private static final Map<String, String> JSON_PATH_EXPRESSION = new HashMap<>();
 
-    private static final Map<String, Function<List<List<Float>>, List<ModelTensor>>> POST_PROCESS_FUNCTIONS = new HashMap<>();
+    private static final Map<String, Function<List<?>, List<ModelTensor>>> POST_PROCESS_FUNCTIONS = new HashMap<>();
 
 
     static {
         JSON_PATH_EXPRESSION.put(OPENAI_EMBEDDING, "$.data[*].embedding");
         JSON_PATH_EXPRESSION.put(COHERE_EMBEDDING, "$.embeddings");
         JSON_PATH_EXPRESSION.put(DEFAULT_EMBEDDING, "$[*]");
-        POST_PROCESS_FUNCTIONS.put(OPENAI_EMBEDDING, buildModelTensorList());
-        POST_PROCESS_FUNCTIONS.put(COHERE_EMBEDDING, buildModelTensorList());
-        POST_PROCESS_FUNCTIONS.put(DEFAULT_EMBEDDING, buildModelTensorList());
+        JSON_PATH_EXPRESSION.put(BEDROCK_EMBEDDING, "$.embedding");
+        POST_PROCESS_FUNCTIONS.put(OPENAI_EMBEDDING, buildMultipleResultModelTensor());
+        POST_PROCESS_FUNCTIONS.put(COHERE_EMBEDDING, buildMultipleResultModelTensor());
+        POST_PROCESS_FUNCTIONS.put(DEFAULT_EMBEDDING, buildMultipleResultModelTensor());
+        POST_PROCESS_FUNCTIONS.put(BEDROCK_EMBEDDING, buildSingleResultModelTensor());
     }
 
-    public static Function<List<List<Float>>, List<ModelTensor>> buildModelTensorList() {
-        return embeddings -> {
+    public static Function<List<?>, List<ModelTensor>> buildSingleResultModelTensor() {
```

Comment:
I didn't clearly understand why do we need two separate functions here?

Could you please explain what's the problem are we facing here?

Reply:
I was intended to separate these two methods to handle different data structure responses, but it's not optimal indeed, I have merged them into one by encapsulating a single dimension list into a list.